.github/workflows/ci-workflow.yml

```diff
@@ -52,12 +52,25 @@ jobs:
           bundle config path vendor/bundle
           bundle update
           bundle install --jobs 4 --retry 3
+      - run: bundle exec rake spec
+
+  engine-lint:
+    needs: engine-specs
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        working-directory: engine
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+          ruby-version: '2.7.4'
+          working-directory: engine
       - run: bundle exec rubocop --format github
```

Comment:
This is to a) allow us to run the demo app job earlier as it's doing more work now b) avoid running lint checks for very single version of ruby/rails. We only care about rspec tests for different versions. We'll need to add this job to the required checks after merging this.